### ISSUE TITLE
Quill removing spaces when tags are adjacent

### ISSUE THREAD

**Issue by ryan-honor-ed** [quilljs/quill#3580]
This only happens when loading the default value (or using dangerouslypastehtml).
If you have two tags that are adjacent, say a "strong" and an anchor "a" and they have a space in between them. so "<strong>please click</strong> <a>here</a>"
then when it gets loaded into a quill editor, the space between these two will get removed.  This is definitely problematic and I haven't found a way around this programmatically.

**Steps for Reproduction**

1. Visit https://codesandbox.io/s/react-uncontrolled-component-default-state-quill-forked-eynydr?file=/src/index.js
2. Click on the "Change Template" button to watch it load the html

**Expected behavior**: It should honor the spaces between tags

**Actual behavior**: You will see the spaces between the different tags all get removed.

**Platforms**: Chrome, React

**Version**: quill 1.3.7

**Comment by easir-kh** [comment]
I experience same issue just with spans `<span>sepa</span> <span>rate</span>` = separate

**Comment by hamurabiaraujo** [comment]
I solved using a workaround, the Zero-width space between the tags. Ex:

`<strong>Lorem</strong> &#x200B;<em>ipsum</em>`

**Comment by matteocarpi** [comment]
I am having a similar issue, though inside the tags. So an initial value of `<p>some text </p>` becomes `<p>some text</p>`. My problem is that I have no control over the initial value, and it's important for me that it stays the same as I am using the editor as part of a form and I need to detect when the value changes from the original to trigger an auto-save.

@ryan-honor-ed did you find a solution in the end? 🙏

**Comment by hamurabiaraujo** [comment]
@matteocarpi, I had a code like this:

`<strong>Free shipping</strong> <em>selected locations</em>`

But the `' '` between the `</strong>` and `<em>` was removed by quill. If a put a content between them, the problem was solved. So how I can't use visible content, used the [Zero-width space](https://en.wikipedia.org/wiki/Zero-width_space) to keep the space. The resultant code:

`<strong>Free shipping</strong> &#x200B;<em>selected locations</em>`

**Comment by matteocarpi** [comment]
Hey @hamurabiaraujo, thanks for the reply!

Ok! so you fixed it manually inserting that in your content. I was wondering if you found a way that would dynamically replace all spaces with a space and the zero-width-space.

**Comment by hamurabiaraujo** [comment]
Yep, in my case a hardcoded solution solved. But I think that you can use a regex or some strategy to see content in the middle of adjacent tags. Then you can join the space + the zero-width.
